File: Ejercicio_0034/Opposite_number.java
package Ejercicio_0034;

import javax.swing.JOptionPane;

public class Opposite_number {
    public static void main(String[] args) {
        String num = JOptionPane.showInputDialog("\nEscribe el numero:\n");

        float num2 = Float.parseFloat(num);

        if (num2 == 0) {
            JOptionPane.showMessageDialog(null, "\nEl 0 no tiene contraparte\n");
        } else if (num2 > 0){
            JOptionPane.showMessageDialog(null, "\nNumero: " + num + "\n\nContraparte: -" + num + "\n");
        } else {
            JOptionPane.showMessageDialog(null, "\nNumero: " + num + "\n\nContraparte: " + num.substring(1) + "\n");
        }
    }
}

File: Ejercicio_0034/Opposite_number.py
def contraparte(n):
    if float(num) == 0:
        print("\nEl 0 no tiene contraparte\n")
    elif float(n) > 0:
        print(f"\n\nNumero: {num}\nContraparte: -{num}\n")
    else:
        print(f"\n\nNumero: {num}\nContraparte: {num[1:]}\n")

num = input("\nEscribe el numero:  ")
contraparte(num)